### ISSUE TITLE
How to get this working with cloud?

### ISSUE THREAD

**Issue by townsmcp** [jacobwgillespie/plex-sync#34]
Is it possible to get Plex-sync working with Plex Cloud?

**Comment by jacobwgillespie** [comment]
Yep, it works just the same as local, you just have to figure out the web address of your remote server.  I'm pretty sure if you find a movie or TV episode, choose "Get Info" from the menu on the left, then click on "View XML" at the bottom of the dialog that pops up, then you will be taken to a page directly using the server hostname.

**Comment by townsmcp** [comment]
I gave that a go last night and managed to get the Google Drive location (seems like it is IPv6 addressing - certainly not IPv4 addressing). I used the following commands though:
./plex-sync XXXXXX@192.168.1.1:32400/7,r https://XXXXXX@XXXXXXXXXXXec3.ams.plex.services/1,w
but I get an error message:
![google drive](https://cloud.githubusercontent.com/assets/14061617/21910530/564d20de-d914-11e6-8480-c6a532275b36.jpg)
Any suggestions?

Also, when I try syncing between my main media server and a readynas, I get a message up saying completed but no changes are actually made (using the same token number and correct library number with 'w' set as the destination. I even tried MATCH_TYPE= precise both both Cloud sync and non cloud sync and got no changes from none cloud and errors on cloud

**Comment by jacobwgillespie** [comment]
Looks like it can't connect to that address at that port - not sure if it helps, but #20 was able to connect to Plex Cloud over https.  Perhaps the combination of https, omitting the port, or another address would help?

**Comment by townsmcp** [comment]
Damn. Still gives out the same error message with or without https and with or without the port (for both my internal and the Google Drive). 
I just double checked the IP and my Plex token and both are correct

**Comment by revog** [comment]
Try using http:// with explicit defining port 80 and the token. 
`http://123456790abcd@ab1234567-c1234567890987654321.ams.plex.services:80/1,w`
This way I was able to sync.

**Comment by townsmcp** [comment]
Thanks for the advice Revog. That worked. The process went through fully. 

After running the code do you do anything with the server instance? Doesn't it tak long for you have have the watched status changed where appropriate? Although I got no error messages a lot of the statuses haven't changed for me in Plex Cloud (using the right user token etc)
Thanks

**Comment by revog** [comment]
@townsmcp: Nice to hear, it's working on your site aswell.
I had the same findings as you wrote. But I think i figured out why this happened. I'm using German Titles. So the titles must be equal on both sides otherwise the matching with the precise flag did not work 100% reliable. I fixed this issue by correcting the assignment.
I don't know yet why the name must be 100% equal, because as far as I know plex-sync should be aware of the imdb id, shouldn't it?

**Comment by jacobwgillespie** [comment]
It should know about IMDb or TMDb ID, yeah, so long as the libraries are set up the same.  The default is fuzzy matching, which compares via an exact match of year, title, and optionally show name, season, etc depending on the type.  Which I'm realizing now that calling it "fuzzy" can be confusing - essentially it compares names, whereas "precise" means the underlying IMDb or TMDb ID has to match.

So, what I'd do is try with the `DRY_RUN` flag so it prints out what it's planning to do, and try with and without the precise matching to see if one is better than the other.  If it's still not doing anything, we can debug from there.  Something like:

```
$ DRY_RUN=1 MATCH_TYPE=precise plex-sync ...

$ DRY_RUN=1 plex-sync ...
```

Should print out a report of items it's going to mark as watched.

**Comment by hashworks** [comment]
Worked for me with forced port 443: `plex-sync https://foobar.ams.plex.services:443/1`

**Comment by FernandoMiguel** [comment]
no luck for me forcing port 443 :(
